docs/src/index.md

```diff
@@ -8,6 +8,76 @@ For more illustrations of its usage in conjunction with other packages, the
 [DataFrames Tutorial using Jupyter Notebooks](https://github.com/bkamins/Julia-DataFrames-Tutorial/)
 is a good complementary resource.
 
+## What is DataFrames?
+
+DataFrames.jl provides a lightweight set of tools for working with tabular data
+in Julia. Its design and functionality are similar to those of `pandas`
+(in Python) and `data.frames` (in R), making it a great general purpose
+data science tool, especially for those coming to Julia from R or Python.
+
+DataFrames plays a central role in the Julia Data ecosystem, and has
+tight integrations with a range of different libraries. DataFrames isn't the
+best tool for everyone working with tabular data --- as noted below, there are
+some other great libraries for certain use-cases --- but it provides great
+data wrangling functionality through a familiar interface.
+
+## DataFrames and the Julia Data Ecosystem
+
+The Julia data ecosystem can be a difficult space for new users to navigate,
+in part because the Julia ecosystem tends to distribute functionality across
+different libraries more than some other languages.
+
+Because many people coming to DataFrames.jl are just starting to explore the
+Julia data ecosystem, below is a list of well-supported libraries that
+provide different data science tools, along with a few notes about
+what makes each library special, and how well integrated they are with
+DataFrames.
+
+- **Statistical Modelling / Machine Learning**
+    - [`Statistics`](https://docs.julialang.org/en/v1/stdlib/Statistics/): The Julia standard library comes with a wide range of statistics functionality, but to gain access to these functions you must first load `Statistics`.
+    - [`FreqTables.jl`](https://github.com/nalimilan/FreqTables.jl): Create frequency tables / cross-tabulations. Tightly integrated with DataFrames.
+    - [`HypothesisTests.jl`](https://juliastats.org/HypothesisTests.jl/stable/): A range of hypothesis testing tools.
+    - [`GLM.jl`](https://juliastats.org/GLM.jl/stable/manual/): Tools for estimating linear and generalized linear models. Tightly integrated with DataFrames.
+    - [`statsmodels.jl`](https://juliastats.org/StatsModels.jl/stable/): For converting heterogeneous DataFrames into homogenous matrices for use with linear algebra libraries or machine learning applications that don't directly support DataFrames. Will do things like convert categorical variables into indicators/one-hot-encodings, create interaction terms, etc.
```

Comment:
```suggestion
    - [`StatsModels.jl`](https://juliastats.org/StatsModels.jl/stable/): For converting heterogeneous DataFrames into homogenous matrices for use with linear algebra libraries or machine learning applications that don't directly support DataFrames. Will do things like convert categorical variables into indicators/one-hot-encodings, create interaction terms, etc.
```
Should make sure that all of the package names are properly capitalized (many of them aren't, won't comment on all)